gitlab_runner_config.py

```diff
@@ -53,20 +56,39 @@ def generate_tags(instance, executor_type="", env=None):
     These tags are specified by runner configs and used by CI specs to run jobs
     on the appropriate host.
     """
-
-    tags = identifying_tags(instance)
-    if executor_type:
-        tags.append(executor_type)
-    if env:
-        tags += [os.environ[e] for e in env if e in os.environ]
+    properties = {
+        "hostname": HOSTNAME,
+        "executor_type": executor_type,
+        "instance": instance,
+        "env": []
+        }
+    if env: 
+        properties["env"] += [os.environ[e] for e in env if e in os.environ]
+    # if executor is batch, gather some more system info for tags
     if executor_type == "batch":
         if which("bsub"):
-            tags.append("lsf")
+            properties["scheduler"] = "lsf"
         elif which("salloc"):
-            tags.append("slurm")
+            properties["scheduler"] = "slurm"
         elif which("cqsub"):
-            tags.append("cobalt")
-    return tags
+            properties["scheduler"] = "cobalt"
+    try:
+        properties.update(capture_tags.capture_tags(instance, executor_type, env=env, tag_schema=tag_schema))
```

Comment:
Definitely the right idea here, but how would the end-user supply their own `capture_tags` method? I was thinking that the end-user could supply (as an argument) a module name that contains the `capture_tags` method. Then at startup in `__main__` you could do

```python
try:
  from userprovidedmodule import capture_tags
except ImportError:
  capture_tags = None # or we could keep a small, default one to provide here instead so we don't have to catch NameError
 ```
```

Reply:
Thank you for getting back to me so quickly. Yes, the way I have written it now, `--capture_tags` is given as a command line argument, which it then tries to import see line 312 of `gitlab_runner_config.py`
```
try:
    capture_tags = __import__(args.capture_tags)
except ModuleNotFoundError:
    logger.info("Tag capture script could not be read.")
```